test/test_check_db_dump_recent.py

```diff
@@ -0,0 +1,217 @@
+
+    def test_recent_success_oracle(self):
+        """Tests the check script given a recent, successful, run."""
+        log_line_list = [
+            "2023-01-01T08:40:01+0000",
+            "2023-01-01T08:41:01+0000 INFO: completed ok",
+            "2023-01-01T09:40:01+0000",
+            "An Error",
+            "2023-01-01T10:40:01+0000",
+            "An Error",
+            "Another Error",
+            "2023-01-01T11:40:01+0000",
+            "2023-01-01T11:41:01+0000 INFO: completed ok",
+        ]
+
+        self._run_test(log_line_list, check.check_db_dump_recent.RETURN_CODE_OK)
+
+    def test_recent_failure_oracle(self):
+        """Tests the check script given a recent, unsuccessful, run."""
+        log_line_list = [
+            "2023-01-01T08:40:01+0000",
+            "2023-01-01T08:41:01+0000 INFO: completed ok",
+            "2023-01-01T09:40:01+0000",
+            "An Error",
+            "2023-01-01T10:40:01+0000",
+            "An Error",
+            "Another Error",
+            "2023-01-01T11:40:01+0000",
+            "2023-01-01T11:41:01+0000 INFO: completed ok",
+            "2023-01-01T11:50:01+0000",
+            "Yet another Error",
+        ]
+
+        self._run_test(log_line_list, check.check_db_dump_recent.RETURN_CODE_OK)
+
+    def test_old_success_oracle(self):
+        """Test the check script given an old, successful, run."""
+        log_line_list = [
+            "2023-01-01T09:40:01+0000",
+            "2023-01-01T09:41:01+0000 INFO: completed ok",
+            "2023-01-01T10:40:01+0000",
+            "An Error",
+            "Another Error",
+            "2023-01-01T11:40:01+0000",
+            "Yet Another Error",
+        ]
+
+        self._run_test(log_line_list, check.check_db_dump_recent.RETURN_CODE_CRITICAL)
+
+    def test_never_successful_oracle(self):
+        """Test the check script given a lack of success, ever."""
+        log_line_list = [
+            "2023-01-01T09:40:01+0000",
+            "An Error",
+            "2023-01-01T10:40:01+0000",
+            "An Error",
+            "Another Error",
+            "2023-01-01T11:40:01+0000",
+            "Yet another Error",
+        ]
+
+        self._run_test(log_line_list, check.check_db_dump_recent.RETURN_CODE_CRITICAL)
+
+    def test_empty_log(self):
+        """Test the check script given an empty input."""
+        log_line_list = []
+
+        self._run_test(log_line_list, check.check_db_dump_recent.RETURN_CODE_CRITICAL)
+
+    def test_no_log(self):
+        """Test the check script given a non existent input."""
+        return_code = self.test_checker.run("/not/a/file")
+
+        self.assertEqual(
+            return_code,
+            check.check_db_dump_recent.RETURN_CODE_CRITICAL,
+            "Expected code %s, but got %s." % (
+                check.check_db_dump_recent.RETURN_CODE_CRITICAL,
+                return_code,
+            ),
+        )
+
+    def list_to_temp_file(self, input_list):
```

Comment:
Not marked as internal, but is this not another internal method like `_run_test`?

Reply:
Fixed in 75199552ead172e431cadc215de6daa462e7bca7